### DISCUSSION TITLE
Calling `reset()` inside `onReset` event doesn't work propely

### DISCUSSION THREAD

**Discussion by vemoo** [react-hook-form/react-hook-form#5454] [Q&A]
**Describe the bug**
When calling `reset()` in the `form`s `onReset` event, the input is cleared but the value is preserved.

**To Reproduce**

[Codesandbox](https://codesandbox.io/s/wizardly-chatelet-pgo4n)

1. Type something in the input, click "Submit", the value shows in the console.
2. Click "Reset", the input is cleared but if we click "Submit" again we see the old value in the console

**Expected behavior**
I would expect the value of the form to be cleared.

**Desktop (please complete the following information):**

- OS: Windows 10 
- Browsers: Chrome
- Version: 91.0.4472.77
also
- Browser: Firefox Developer Edition
- Version: 90.0b1
also
- Browser: Microsoft Edge
- Version: 91.0.864.37

**Additional context**
If I change 
```js
const onReset = () => {
    reset();
};
```
to
```js
const onReset = () => {
    setTimeout(reset);
};
```
~it works as expected.~ (see next comment)

**Comment by vemoo** [comment]
I just realised that using `setTimeout(reset)` causes an infinite render loop, so it doesn't really work.

**Comment by bluebill1049** [comment]
https://react-hook-form.com/api/useform/reset/
<img width="1049" alt="Screen Shot 2021-06-02 at 9 13 44 pm" src="https://user-images.githubusercontent.com/10513364/120470892-6b4cf780-c3e7-11eb-9a7d-4266229eb1ba.png">

**Comment by vemoo** [comment]
But that means that I have to specify the default value of all fields, doesn't it? Because `defaultValues: {}` doesn't work.

If that's the desired behaviour then my issue was that for a `type=reset` button the browser itself clears the fields, but `react-hook-form` doesn't know about it so it ends up out of sync. The solution is to call `preventDefault` on the reset event, maybe that's something worth mentioning in the docs.
